src/AssetStorage.mo

```diff
@@ -61,7 +61,7 @@ module AssetStorage {
     public type StreamingStrategy = {
         #Callback : {
             token : StreamingCallbackToken;
-            callback : shared query StreamingCallbackToken -> async StreamingCallbackHttpResponse;
+            callback : shared () -> async ();
```

Comment:
> You can trick the compiler to send a different reference type.

Is this a bug in the compiler or intended behaviour?

Reply:
it's a bug
https://github.com/dfinity/candid/issues/273